### ISSUE TITLE
'Extract Files' task for Release fails

### ISSUE THREAD

**Issue by atul898** [Microsoft/vsts-tasks#7243]
## Environment
- Server - VSTS

- Agent - Hosted

## Issue Description

Extract Files task for Release fails

### Error logs

2018-05-17T21:33:35.1512744Z ##[error]Extraction failed for file: C:\VSTS_DeploymentAgent\A1\_work\r1\a\_Project-ASP.NET-CI\drop\ABCProject.Web.zip 
2018-05-17T21:33:35.1522566Z 
2018-05-17T21:33:35.1522820Z ERROR: Can't allocate required memory!

**Comment by bryanmacfarlane** [comment]
@stephenmichaelf - sounds familiar?

**Comment by stephenmichaelf** [comment]
Yes. @atul898 We have seen in the past this is caused by the destination files already existing. Do you clear your drop folder before starting?

@bryanmacfarlane The PR to fix this is [here](https://github.com/Microsoft/vsts-tasks/pull/6760). I thought I had made the FF change but it looks like I didn't or didn't push it. Will get this wrapped up and merged today.

**Comment by lukewis** [comment]
@stephenmichaelf Is this still on the radar?  Looks like the pull request was approved, just needs some merge work?

**Comment by shawncrabtree** [comment]
This would be nice to have fixed. Currently we have multiple test projects that we're extracting and they share some dependencies. I think trying to extract these duplicated dependent dlls to the same output directory is what's causing this error because they can't be overwritten. Our workaround is creating an extraction stage for each of our projects, that each output to a different directory.

**Comment by raswinraaj** [comment]
Is this bug going to be fixed and merged? It looks like a long pending open issue. Any work arounds on this?

**Comment by mikariekkinen** [comment]
My workaround is to use Bash task and do the unzipping with
`unzip -qo mystuff.zip`

**Comment by Benalink** [comment]
Just came across this issue today, release always fails now as it prompts for input before overwriting each file but with the error message after the prompt reading `ERROR: Can't allocate required memory!`

Will switch to a shell command or something that can handle overwriting files for now but it would be nice if this fix got merged in at some point.

**Comment by Asier-Villanueva** [comment]
I get this error in Azure DevOps Release Pipeline with the check "Clean destination folder before extracting" checked.

Log:

> 2020-03-09T16:50:23.8568707Z Extracting archive: d:\a\r1\a\_Task Runner WebJob Debug CI\TaskRunner\TaskRunner.zip
> 2020-03-09T16:50:23.8569119Z --
> 2020-03-09T16:50:23.8569537Z Path = d:\a\r1\a\_Task Runner WebJob Debug CI\TaskRunner\TaskRunner.zip
> 2020-03-09T16:50:23.8570269Z Type = zip
> 2020-03-09T16:50:23.8570569Z Physical Size = 35662565
> 2020-03-09T16:50:23.8570795Z 
> 2020-03-09T16:50:23.8570980Z 
> 2020-03-09T16:50:23.8571283Z Would you like to replace the existing file:
> 2020-03-09T16:50:23.8571672Z   Path:     d:\a\r1\a\TaskRunnerFiles\Azure.Core.dll
> 2020-03-09T16:50:23.8572063Z   Size:     123256 bytes (121 KiB)
> 2020-03-09T16:50:23.8572403Z   Modified: 2020-02-10 16:45:50
> 2020-03-09T16:50:23.8572748Z with the file from archive:
> 2020-03-09T16:50:23.8573068Z   Path:     Azure.Core.dll
> 2020-03-09T16:50:23.8573391Z   Size:     123256 bytes (121 KiB)
> 2020-03-09T16:50:23.8577344Z   Modified: 2020-02-10 16:45:50
> 2020-03-09T16:50:23.8577793Z ? (Y)es / (N)o / (A)lways / (S)kip all / A(u)to rename all / (Q)uit? 
> 2020-03-09T16:50:23.8578226Z Archives with Errors: 1
> 2020-03-09T16:50:23.8578482Z 
> 2020-03-09T16:50:23.8580380Z 
> 2020-03-09T16:50:23.8580956Z ERROR: Can't allocate required memory!
> 2020-03-09T16:50:23.8581743Z Error: Extraction failed for file: d:\a\r1\a\_Task Runner WebJob Debug CI\TaskRunner\TaskRunner.zip 
> 2020-03-09T16:50:23.8582574Z code: 2 
> 2020-03-09T16:50:23.8634534Z ##[error]Extraction failed for file: d:\a\r1\a\_Task Runner WebJob Debug CI\TaskRunner\TaskRunner.zip 
> code: 2

**Comment by Miles-Davies-HORIBA** [comment]
I am currently suffering from this issue also. I will work around in the shell but this issue should really be fixed. Open for 2 years now :-(

**Comment by AlexA-1976** [comment]
I'm also seeing this issue, trying to extract multiple UI test projects into the same destination folder.

**Comment by raswinraaj** [comment]
As a workaround, I moved the extract and the subsequent tasks to a different agent job. That solved the issue.

**Comment by Miles-Davies-HORIBA** [comment]
poke

**Comment by b926vb** [comment]
We are facing the same issue with this task in a deployment job. When will this be resolved?
